concurrent/src/test/java/com/facebook/airlift/concurrent/TestConcurrentScheduledExecutor.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.airlift.concurrent;
+
+import org.testng.annotations.Test;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestConcurrentScheduledExecutor
```

Comment:
This class doesn't test a lot of the methods. Not sure how picky we want to be, but generally I am pretty pedantic about unit testing even really simple methods for utility classes like these.

Reply:
Because they have the same pattern: 1) get a scheduler 2) submit task,  I only test a few representative ones.